test/libevmasm/PlainAssemblyParser.h

```diff
@@ -0,0 +1,79 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libsolutil/JSON.h>
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace solidity::evmasm::test
+{
+
+/// Parser for the plain assembly format. The format is meant to be good enough for humans to read
+/// while being sstraightforward to map the assembly JSON format that solc can import.
+///
+/// Syntax:
+/// - Every line consists of zero or more whitespace-separated tokens.
+/// - A token that begins with `//` starts a comment, which extends to the end of the line.
+/// - A non-empty line represents a single assembly item.
+/// - The name of the item is the first thing on the line and may consist of one or more tokens.
+/// - One or more arguments follow the name.
+///
+/// Supported items:
+/// - All instruction names.
+/// - PUSH <hex value>
+/// - PUSH [tag] <tagID>
+/// - tag <tagID>
+class PlainAssemblyParser
+{
+public:
+	/// Parses plain assembly format and returns the equivalent assembly  JSON.
+	/// Errors are reported by throwing runtime_error.
+	Json parse(std::string _sourceName, std::string const& _source);
+
+protected:
```

Comment:
why did you chose protected here, do you expect to inherit from the parser?

Reply:
Because it's something between the public and the private part. It's safe to expose to the user, unlike direct access to member variables, but it's also more low-level than `parse()`. I don't necessarily plan to inherit from it but if I did, that part would be what I'd expose to the derived class, so `protected` seemed fitting.

Reply:
Makes sense if you put it like that, I would've probably just crammed everything into `private` scope. Thanks!